src/lib/infrastructure/server/gateway/openai-agent-gateway.ts

```diff
@@ -26,44 +27,46 @@ export default class OpenAIAgentGateway implements AgentGatewayOutputPort<TOpenA
   ) {
     this.logger = loggerFactory("OpenAIAgentGateway");
   }
-  async createAgent(researchContextID: number, researchContextName: string, researchContextDesciption: string, vectorStoreID: string): Promise<TCreateAgentDTO> {
-    const kpCredentialsDTO = await this.AuthGateway.extractKPCredentials();
-    if (!kpCredentialsDTO.success) {
-      this.logger.error({ kpCredentialsDTO }, "Failed to extract KP credentials from session");
-      return {
-        success: false,
-        data: {
-          message: "Failed to extract KP credentials from session",
-          operation: "openai:create-agent",
-        },
-      };
-    }
-    const clientID = kpCredentialsDTO.data.clientID;
-    const agentName = generateAgentName(clientID, researchContextID);
-    const instructions = `You are an expert data analyst specialized in ${researchContextName}. Your research context can be best described by ${researchContextDesciption}. You will help me and my team explore and analize some datasets that we have augmented, by combining data from satellites, twitter, and telegram, regarding the occurrence of disaster events related to ${researchContextName} at different locations. You have access to a code interpreter to generate insights from the data, and a file search tool to find relevant datasets.`;
+
+  async createAgent(researchContextTitle: string, researchContextDescription: string, vectorStoreID: string, additionalFiles?: RemoteFile[]): Promise<TCreateAgentDTO> {
+    const instructions = `You are an expert data analyst specialized working in the research context with title '${researchContextTitle}'. This research context has the following description '${researchContextDescription}'. You have also been assigned some files. Some of these are images in different formats (e.g., JPG, JPEG, PNG, etc.), assigned to you via normal code interpreter. Other files have been assigned to you via a vector store. So please consider this anytime the user asks you about files you have access to.
+    You will help me and my team explore and analize some datasets that we have augmented, by combining data from satellites, twitter, and telegram, regarding the occurrence of disaster events related to '${researchContextTitle}' at different locations. You have access to a code interpreter to generate insights from the data, and a file search tool to find relevant datasets.`; // TODO: discuss this
```

Comment:
@maany I was playing with this a a bit, but might still need some tweaking

Reply:
@maany more context: sometimes it doesnt respond directly about files in the vector store until asked explicitly about one of them in particular (e.g., saying "look at the .js file" or "at the blabla.extension file")

![imagen](https://github.com/user-attachments/assets/697a028f-5165-4ad6-8f24-96387a1f12fe)